### ISSUE TITLE
Geofencing fancy notifications

### ISSUE THREAD

**Issue by flibbertigibbet** [azavea/cac-tripplanner-android#46]
When geofence triggers and notification displays, add a deep link to detail view for the place.

Also possibly ask if user is at the place, and if so, change the flag to "been" and remove the geofence.

@lederer might have some other ideas on how this interaction should work.

**Comment by lederer** [comment]
I'm thinking something like this…

Whenever we think you're at/near a place, we show a special dialog floating at the bottom of every screen of the app, that says:

---

Are you at **John Heinz National Wildlife Refuge**?
[ Yes ] [ No ]

---

Tapping _Yes_ dismisses the dialog and changes the flag to been. Tapping _No_ dismisses the dialog without changing the flag.

Since the dialog would appear on every screen, then by implication it would appear on the detail view for that place and so you'd see it when you follow the notification's deep link. So they could just tap Yes (or No) once they land there.

How does that sound?

**Comment by maurizi** [comment]
> we show a special dialog floating at the bottom of every screen of the app, that says

Wouldn't we prefer to show the notification using the system notification area, not in-app?

**Comment by lederer** [comment]
> Wouldn't we prefer to show the notification using the system notification area, not in-app?

I don't think so. Here's why…

We're trying to make the notification serve two distinct use cases: 

- **_Nearby_**: Hey, this place is nearby. Care to check it out?
- **_There_**: Hey, this place is nearby. And in fact geolocation accuracy can be wonky enough that you might even be there right now. In which case, care to add it to your "been" list?

I think the way to finesse that duality is to sequence it rather than overload the notification. That is, make **_Nearby_** the primary, unambiguous purpose of the notification, free of the noise and complexity of the yes/no actions for **_There_**, while allowing the notification to also serve as a natural entry point[1] to the app for **_There_**, where we can offer an unambiguous UI for that.

[1] Perhaps enhanced with some clever notification message copy that signals them to tap into the app  in the case of **_There_**.

**Comment by flibbertigibbet** [comment]
We're going to go with just a 'been' button in the notification for the scope of this issue.

**Comment by lederer** [comment]
Here's the design for the detail screen when at a place:
https://marvelapp.com/4c295bb/screen/44571061

![image](https://user-images.githubusercontent.com/128699/41663611-d4eb4dcc-7471-11e8-8229-28738830c299.png)

Tapping "I'm here now" marks the place as `been`. The UI should then appear just as it normally would  when the user has `been` to that place.

Tapping the X restores the component to its default format ("Been here? Want to go?", etc).

**Comment by flibbertigibbet** [comment]
@lederer thanks for the mockup, could you put it in a separate issue please?
